workshop/3-Experimentation/3.5-Amplitude-Performance-Metrics.ipynb

```diff
@@ -0,0 +1,461 @@
+    "\n",
+    "Speaking of organic searches, let's look at how personalized search results ranking performed against regular searches."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Amplitude Search Personalization Conversion Reports\n",
+    "\n",
+    "The Retail Demo Store allows users to perform a text search for products.  In the earlier part of the workshop, you deployed Personalize to re-sort product searches to show users products that might be more relevant based on their browsing history.\n",
+    "\n",
+    "When the user performs a search in the Retail Demo Store, a ProductSearched event is sent:\n",
+    "\n",
+    "```\n",
+    "if (this.amplitudeEnabled()) {\n",
+    "    // Amplitude event\n",
+    "    var eventProperties = {\n",
+    "        query: query,\n",
+    "        reranked: (user ? 'true' : 'false'),\n",
+    "        resultCount: numResults\n",
+    "    };\n",
+    "    Amplitude.getInstance().logEvent('ProductSearched', eventProperties);\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "If the search results are re-ranked by Personalize, the `reranked` property will be set to True in the event.  This means that we can compare user funnels from before and after Personalize is deployed.\n",
+    "\n",
+    "For this funnel report, you will create an Amplitude funnel report that looks at:\n",
+    "\n",
+    "ProductSearched >\n",
+    "ProductAdded >\n",
+    "CheckoutStarted >\n",
+    "OrderCompleted\n",
+    "\n",
+    "First, let's look at search conversion before Personalize is deployed.  Set up a funnel report, with the events shown above.\n",
+    "\n",
+    "![](images/amplitude/search-funnel-pre-setup.png)\n",
+    "\n",
+    "Then select a time range that between 60 and 30 days before today (this is the time period prior to personalization being deployed in your dataset).\n",
+    "\n",
+    "![](images/amplitude/search-funnel-pre-dates.png)\n",
+    "\n",
+    "You should see a report that looks like this.\n",
+    "\n",
+    "![](images/amplitude/search-funnel-pre-pers.png)\n",
+    "\n",
+    "In this case, there is a 43% conversion rate for organic searches without personalized ranking.\n",
+    "\n",
+    "Now, let's compare these results to search conversion after Personalize is deployed to re-rank search results.  In your funnel report, select the `where` option for the `ProductSearched` event, and then select the `reranked` property.  Select the `True` option.\n",
+    "\n",
+    "![](images/amplitude/search-funnel-post-setup.png)\n",
+    "\n",
+    "Then, select a time frame for the last 30 days, since this how long your dataset has personalization event data.\n",
+    "\n",
+    "![](images/amplitude/search-funnel-post-dates.png)\n",
+    "\n",
+    "Your report should now show conversions only for personalized events.\n",
+    "\n",
+    "![](images/amplitude/search-funnel-post.png)\n",
+    "\n"
+   ]
+  }
```

Comment:
Minor suggestion: add closing statement if nothing else to indicate that the workshop has been completed.